addons/mrp/models/mrp_production.py

```diff
@@ -1226,13 +1226,15 @@ def action_view_mrp_production_backorders(self):
     def _prepare_stock_lot_values(self):
         self.ensure_one()
         name = self.env['ir.sequence'].next_by_code('stock.lot.serial')
-        exist_lot = self.env['stock.lot'].search([
+        next_serial = not name or self.env['stock.lot'].search([
             ('product_id', '=', self.product_id.id),
             ('company_id', '=', self.company_id.id),
             ('name', '=', name),
         ], limit=1)
-        if exist_lot:
+        if next_serial:
             name = self.env['stock.lot']._get_next_serial(self.company_id, self.product_id)
+        if not name:
+            raise UserError(_("Please set the first Serial Number or a default sequence"))
```

Comment:
Could also set a default SN like "0000000" instead of throwing an error

Reply:
never mind the user had to manually disable the default sequence so better raise the error